diff -Nru Vuze_5700_source/org/gudy/azureus2/ui/swt/mainwindow/SWTThread.java Vuze_5700_source/org/gudy/azureus2/ui/swt/mainwindow/SWTThread.java.daviddavid
--- Vuze_5700_source/org/gudy/azureus2/ui/swt/mainwindow/SWTThread.java	2015-11-06 23:04:34.000000000 +0100
+++ Vuze_5700_source/org/gudy/azureus2/ui/swt/mainwindow/SWTThread.java.daviddavid	2015-11-16 09:34:12.485713620 +0100
@@ -226,66 +226,6 @@
 			}
 		});
 
-		if (Constants.isOSX) {
-			
-			// On Cocoa, we get a Close trigger on display.  Need to check if all
-			// platforms send this.
-			display.addListener(SWT.Close, new Listener() {
-				public void handleEvent(Event event) {
-					UIFunctions uiFunctions = UIFunctionsManager.getUIFunctions();
-					if (uiFunctions != null) {
-						event.doit = uiFunctions.dispose(false, false);
-					}
-				}
-			});
-
-			String platform = SWT.getPlatform();
-			// use reflection here so we decouple generic SWT from OSX specific stuff to an extent
-
-			if (platform.equals("carbon")) {
-				try {
-
-					Class<?> ehancerClass = Class.forName("org.gudy.azureus2.ui.swt.osx.CarbonUIEnhancer");
-
-					Constructor<?> constructor = ehancerClass.getConstructor(new Class[] {});
-
-					constructor.newInstance(new Object[] {});
-
-				} catch (Throwable e) {
-
-					Debug.printStackTrace(e);
-				}
-			} else if (platform.equals("cocoa")) {
-				try {
-
-					Class<?> ehancerClass = Class.forName("org.gudy.azureus2.ui.swt.osx.CocoaUIEnhancer");
-
-					Method mGetInstance = ehancerClass.getMethod("getInstance", new Class[0]);
-					Object claObj = mGetInstance.invoke(null, new Object[0] );
-
-					Method mHookAppMenu = claObj.getClass().getMethod("hookApplicationMenu", new Class[] {});
-					if (mHookAppMenu != null) {
-						mHookAppMenu.invoke(claObj, new Object[0]);
-					}
-
-					Method mHookDocOpen = claObj.getClass().getMethod("hookDocumentOpen", new Class[] {});
-					if (mHookDocOpen != null) {
-						mHookDocOpen.invoke(claObj, new Object[0]);
-					}
-					
-					Method mIsRetinaDisplay = claObj.getClass().getMethod("isRetinaDisplay");
-					if (mIsRetinaDisplay != null) {
-						isRetinaDisplay = (Boolean) mIsRetinaDisplay.invoke(claObj);
-					}
-
-					
-				} catch (Throwable e) {
-
-					Debug.printStackTrace(e);
-				}
-			}
-		}   
-
 		if (app != null) {
 			app.runInSWTThread();
 			runner = new Thread(new AERunnable() {
