diff -up Vuze_5300_source/build.xml Vuze_5300_source/build.xml.java5
--- Vuze_5300_source/build.xml	2014-03-01 13:06:03.265477792 +0100
+++ Vuze_5300_source/build.xml.java5	2014-03-01 15:20:43.648952667 +0100
@@ -36,7 +36,7 @@ NOTE:  You may need to set the  ANT_OPTS
          <fileset dir="${root.dir}/${libs.dir}" includes="**/*.jar" />
       </path>
    
-   	<javac srcdir="${root.dir}" destdir="${root.dir}" nowarn="yes" source="1.4" target="1.4" includeAntRuntime="no" debug="true" debuglevel="lines,vars,source" >
+   	<javac encoding="8859_1" srcdir="${root.dir}" destdir="${root.dir}" nowarn="yes" includeAntRuntime="no" debug="true" debuglevel="lines,vars,source" >
          <classpath refid="libs.classpath" />
       </javac>
    </target>
