Index: vuze-4.3.0.4-disable-updates/com/aelitis/azureus/ui/swt/Initializer.java
===================================================================
--- vuze-4.3.0.4-disable-updates/com/aelitis/azureus/ui/swt/Initializer.java
+++ vuze-4.3.0.4-disable-updates/com/aelitis/azureus/ui/swt/Initializer.java	2009-12-09 04:00:23.543846143 +0200
@@ -44,8 +44,6 @@
 import org.gudy.azureus2.ui.swt.pluginsinstaller.InstallPluginWizard;
 import org.gudy.azureus2.ui.swt.progress.ProgressWindow;
 import org.gudy.azureus2.ui.swt.update.UpdateMonitor;
-import org.gudy.azureus2.ui.swt.updater2.PreUpdateChecker;
-import org.gudy.azureus2.ui.swt.updater2.SWTUpdateChecker;
 
 import com.aelitis.azureus.core.*;
 import com.aelitis.azureus.core.cnetwork.ContentNetwork;
@@ -359,11 +357,6 @@
 					
 					reportCurrentTaskByKey("splash.openViews");
 	
-					SWTUpdateChecker.initialize();
-	
-					PreUpdateChecker.initialize(core,
-							COConfigurationManager.getStringParameter("ui"));
-	
 					UpdateMonitor.getSingleton(core); // setup the update monitor
 	
 					//Tell listeners that all is initialized :
Index: vuze-4.3.0.4-disable-updates/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java
===================================================================
--- vuze-4.3.0.4-disable-updates/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java
+++ vuze-4.3.0.4-disable-updates/org/gudy/azureus2/pluginsimpl/local/PluginInitializer.java	2009-12-09 04:00:23.544845540 +0200
@@ -122,24 +122,6 @@
 					"Magnet URI Handler",
 					"true",
 					"false"},
-			{	 PluginManagerDefaults.PID_CORE_UPDATE_CHECKER, 
-   					"org.gudy.azureus2.update.CoreUpdateChecker", 
-   					"azbpcoreupdater", 
-   					"CoreUpdater",
-					"true",
-					"true"},
-			{	 PluginManagerDefaults.PID_CORE_PATCH_CHECKER, 
-   					"org.gudy.azureus2.update.CorePatchChecker", 
-   					"azbpcorepatcher", 
-   					"CorePatcher",
-					"true",
-					"true"},
-	   		{	 PluginManagerDefaults.PID_PLATFORM_CHECKER, 
-   					"org.gudy.azureus2.platform.PlatformManagerPluginDelegate", 
-   					"azplatform2", 
-   					"azplatform2",
-					"true",
-					"false"},
 	   		//{	 PluginManagerDefaults.PID_JPC, 
 				//	"com.aelitis.azureus.plugins.jpc.JPCPlugin", 
 				//	"azjpc", 
Index: vuze-4.3.0.4-disable-updates/org/gudy/azureus2/pluginsimpl/update/PluginUpdatePlugin.java
===================================================================
--- vuze-4.3.0.4-disable-updates/org/gudy/azureus2/pluginsimpl/update/PluginUpdatePlugin.java
+++ vuze-4.3.0.4-disable-updates/org/gudy/azureus2/pluginsimpl/update/PluginUpdatePlugin.java	2009-12-09 04:00:23.544845540 +0200
@@ -47,7 +47,6 @@
 import org.gudy.azureus2.plugins.ui.model.*;
 import org.gudy.azureus2.pluginsimpl.*;
 import org.gudy.azureus2.pluginsimpl.update.sf.*;
-import org.gudy.azureus2.update.CorePatchChecker;
 
 import com.aelitis.azureus.core.versioncheck.VersionCheckClient;
 
@@ -376,6 +375,12 @@
 				
 				PluginInterface	pi = plugins[i];
 				
+				String plugdir = pi.getPluginDirectoryName();
+				if (plugdir == null || plugdir.startsWith("/usr/share")) {
+					// system wide installation
+					continue;
+				}
+
 				if ( pi.getPluginState().isDisabled()){
 					
 						// if it is disabled because it failed to load, carry on and check for updates as the newer version
@@ -833,14 +838,8 @@
 				
 					// update to a built-in plugin
 				
-				log.log( LoggerChannel.LT_INFORMATION, "    This is a built-in plugin, updating core" );
+				log.log( LoggerChannel.LT_INFORMATION, "    This is a built-in plugin; update skipped (system-wide install); this update should not have been downloaded, this is a bug" );
 
-				CorePatchChecker.patchAzureus2( update.getCheckInstance(), data, plugin.getPluginID() + "_" + version, log );
-				
-					// always need to restart for this
-				
-				update.setRestartRequired( Update.RESTART_REQUIRED_YES );
-				
 			}else{
 
 				final File	plugin_dir	= new File( plugin_dir_name );
Index: vuze-4.3.0.4-disable-updates/org/gudy/azureus2/ui/console/ConsoleInput.java
===================================================================
--- vuze-4.3.0.4-disable-updates/org/gudy/azureus2/ui/console/ConsoleInput.java
+++ vuze-4.3.0.4-disable-updates/org/gudy/azureus2/ui/console/ConsoleInput.java	2009-12-09 04:00:23.545844658 +0200
@@ -59,7 +59,6 @@
 import org.gudy.azureus2.ui.common.UIConst;
 import org.gudy.azureus2.ui.console.commands.*;
 import org.gudy.azureus2.ui.console.util.TextWrap;
-import org.gudy.azureus2.update.CorePatchChecker;
 import org.gudy.azureus2.update.UpdaterUpdateChecker;
 
 import com.aelitis.azureus.core.AzureusCore;
@@ -699,14 +698,7 @@
 					}
 			});
 		
-		PluginInterface	pi = pm.getPluginInterfaceByClass( CorePatchChecker.class );
-		
-		if ( pi != null ){
-			
-			pi.getPluginState().setDisabled( true );
-		}
-		
-		pi = pm.getPluginInterfaceByClass( UpdaterUpdateChecker.class );
+		PluginInterface pi = pm.getPluginInterfaceByClass( UpdaterUpdateChecker.class );
 		
 		if ( pi != null ){
 			
Index: vuze-4.3.0.4-disable-updates/org/gudy/azureus2/ui/swt/update/UpdateMonitor.java
===================================================================
--- vuze-4.3.0.4-disable-updates/org/gudy/azureus2/ui/swt/update/UpdateMonitor.java
+++ vuze-4.3.0.4-disable-updates/org/gudy/azureus2/ui/swt/update/UpdateMonitor.java	2009-12-09 04:02:15.698844843 +0200
@@ -40,7 +40,6 @@
 import org.gudy.azureus2.ui.swt.progress.IProgressReporter;
 import org.gudy.azureus2.ui.swt.progress.IProgressReporterListener;
 import org.gudy.azureus2.ui.swt.progress.ProgressReportingManager;
-import org.gudy.azureus2.update.CoreUpdateChecker;
 
 import com.aelitis.azureus.core.AzureusCore;
 import com.aelitis.azureus.ui.*;
@@ -170,55 +169,6 @@
 					public void
 					run()
 					{
-						// check for non-writeable app dir on non-vista platforms (vista we've got a chance of
-						// elevating perms when updating) and warn user. Particularly useful on OSX when
-						// users haven't installed properly
-		
-						if ( !( Constants.isWindowsVistaOrHigher || SystemProperties.isJavaWebStartInstance())){
-		
-							String	app_str = SystemProperties.getApplicationPath();
-		
-							if ( !new File(app_str).canWrite()){
-		
-								final UIFunctions uiFunctions = UIFunctionsManager.getUIFunctions();
-		
-								if ( uiFunctions != null ){
-		
-									if ( app_str.endsWith( File.separator )){
-		
-										app_str = app_str.substring(0, app_str.length()-1);
-									}
-									
-									final String f_app_str = app_str;
-									
-									Utils.execSWTThread(
-										new Runnable()
-										{
-											public void
-											run()
-											{
-												UIFunctionsUserPrompter prompt = 
-													uiFunctions.getUserPrompter(
-														MessageText.getString("updater.cant.write.to.app.title"), 
-														MessageText.getString("updater.cant.write.to.app.details", new String[]{f_app_str}), 
-														new String[]{ MessageText.getString( "Button.ok" )}, 
-														0 );
-				
-												//prompt.setHtml( "http://a.b.c/" );
-					
-												prompt.setIconResource( "warning" );
-					
-												prompt.setRemember( "UpdateMonitor.can.not.write.to.app.dir.2", false, 
-														MessageText.getString( "MessageBoxWindow.nomoreprompting" ));
-					
-												prompt.open(null);
-											}
-										},
-										true );
-								}
-							}
-						}
-		
 						performAutoCheck(true);
 					}
 				});
@@ -413,8 +363,6 @@
 							uiFunctions.setStatusText("");
 						}
 					}
-
-					CoreUpdateChecker.doUsageStats();
 				}
 			});
 		}
Index: vuze-4.3.0.4-disable-updates/org/gudy/azureus2/ui/swt/mainwindow/Initializer.java
===================================================================
--- vuze-4.3.0.4-disable-updates/org/gudy/azureus2/ui/swt/mainwindow/Initializer.java
+++ vuze-4.3.0.4-disable-updates/org/gudy/azureus2/ui/swt/mainwindow/Initializer.java	2009-12-09 04:00:23.546843495 +0200
@@ -45,8 +45,6 @@
 import org.gudy.azureus2.ui.swt.pluginsinstaller.InstallPluginWizard;
 import org.gudy.azureus2.ui.swt.progress.ProgressWindow;
 import org.gudy.azureus2.ui.swt.update.UpdateMonitor;
-import org.gudy.azureus2.ui.swt.updater2.PreUpdateChecker;
-import org.gudy.azureus2.ui.swt.updater2.SWTUpdateChecker;
 
 import com.aelitis.azureus.core.*;
 import com.aelitis.azureus.core.util.CopyOnWriteList;
@@ -254,10 +252,6 @@
 						if (finalLogListener != null)
 							Logger.removeListener(finalLogListener);
 	
-						SWTUpdateChecker.initialize();
-	
-						PreUpdateChecker.initialize( core, COConfigurationManager.getStringParameter("ui"));
-	
 						UpdateMonitor.getSingleton(core); // setup the update monitor
 	
 						//Tell listeners that all is initialized :
